test/Microsoft.AspNetcore.CertificateGeneration.Task.Tests/GenerateSSLCertificateTaskTest.cs

```diff
@@ -0,0 +1,226 @@
+        }
+
+        public X509CertificateCollection GetTestCertificates()
+        {
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadWrite);
+                var certificates = store.Certificates.Find(X509FindType.FindBySubjectDistinguishedName, TestSubject, validOnly: false);
+                store.Close();
+
+                return certificates;
+            }
+        }
+
+        private void EnsureCleanUp()
+        {
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadWrite);
+                var certificates = store.Certificates.Find(X509FindType.FindBySubjectDistinguishedName, TestSubject, validOnly: false);
+                store.RemoveRange(certificates);
+                store.Close();
+            }
+        }
+
+        public void Dispose()
+        {
+            EnsureCleanUp();
+        }
+
+        private void CreateCertificate(
+            DateTimeOffset notBefore = default(DateTimeOffset),
+            DateTimeOffset expires = default(DateTimeOffset),
+            bool savePrivateKey = true)
+        {
+            using (var rsa = RSA.Create(2048))
+            {
+                var signingRequest = new CertificateRequest(
+                    new X500DistinguishedName(TestSubject), rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+                var enhacedKeyUsage = new OidCollection();
+                enhacedKeyUsage.Add(new Oid("1.3.6.1.5.5.7.3.1", "Server Authentication"));
+                signingRequest.CertificateExtensions.Add(new X509EnhancedKeyUsageExtension(enhacedKeyUsage, critical: true));
+                signingRequest.CertificateExtensions.Add(new X509KeyUsageExtension(X509KeyUsageFlags.KeyEncipherment, critical: true));
+                signingRequest.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(
+                        certificateAuthority: false,
+                        hasPathLengthConstraint: false,
+                        pathLengthConstraint: 0,
+                        critical: true));
+
+                var sanBuilder = new SubjectAlternativeNameBuilder();
+                sanBuilder.AddDnsName(TestSubject.Replace("CN=", ""));
+                signingRequest.CertificateExtensions.Add(sanBuilder.Build());
+
+                var certificate = signingRequest.CreateSelfSigned(
+                    notBefore == default(DateTimeOffset) ? DateTimeOffset.Now : notBefore,
+                    expires == default(DateTimeOffset) ? DateTimeOffset.Now.AddYears(1) : expires);
+
+                if (savePrivateKey)
+                {
+                    var export = certificate.Export(X509ContentType.Pkcs12, "");
+                    var imported = new X509Certificate2(export, "", X509KeyStorageFlags.PersistKeySet);
```

Comment:
Need to mark the cert as exportable. `X509KeyStorageFlags.PersistKeySet | X509KeyStorageFlags.Exportable`